Application/src/Model/diagramObject.java

```diff
@@ -0,0 +1,215 @@
+package Model;
+
+import com.google.gson.annotations.SerializedName;
+import java.util.List;
+
+public class diagramObject {
+
+
+
+
+    //handle diagram's meta
+    private Meta meta;
+
+    public Meta getMeta() {
+        return meta;
+    }
+
+    public void setMeta(Meta meta) {
+        this.meta = meta;
+    }
+
+
+    //handle the Type of the diagram
+    private String type;
+
+    public void setType(String type) {
+        this.type = type;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+
+    //handle the Processes of the diagram
+    private List<Processes> processes = null;
+
+    public void setProcesses(List<Processes> processes) {
+        this.processes = processes;
+    }
+
+    public List<Processes> getProcesses() {
+        return processes;
+    }
+
+
+    //handle the messages of the diagram
+    private Diagram diagram;
+
+    public Diagram getDiagram() {
+        return diagram;
+    }
+
+    public void setDiagram(Diagram diagram) {
+        this.diagram = diagram;
+    }
+
+
+
+    public class Meta {
+
+        private String format;
+        private String version;
+
+        private List<Object> extensions = null;
+
+        private void setExtensions(List<Object> extensions) {
+            this.extensions = extensions;
+        }
+
+        public List<Object> getExtensions() {
+            return extensions;
+        }
+
+
+        public String getFormat() {
+            return format;
+        }
+        public void setFormat(String format) {
+            this.format = format;
+        }
+        public String getVersion() {
+            return version;
+        }
+        public void setVersion(String version) {
+            this.version = version;
+        }
+
+
+    }
+
+    public class Processes {
+
+
+        @SerializedName("class")
+        private String class1;
```

Comment:
class1?????

Reply:
we can't call it class because the word class is protected in java, the solution was to give it a name for serialization purposes i.e. line 95
This annotation will make the gson library see "class" instead of "class1" when matching and parsing.